delocate/tools.py

```diff
@@ -363,25 +590,22 @@ def get_rpaths(filename):
     rpath : tuple
         rpath paths in `filename`
     """
-    try:
-        lines = _cmd_out_err(["otool", "-l", filename])
-    except RuntimeError:
-        return ()
-    if not _line0_says_object(lines[0], filename):
+    otool = subprocess.run(
+        ["otool", "-l", filename],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    if not _line0_says_object(otool.stdout or otool.stderr, filename):
         return ()
-    lines = [line.strip() for line in lines]
-    paths = []
-    line_no = 1
-    while line_no < len(lines):
-        line = lines[line_no]
-        line_no += 1
-        if line != "cmd LC_RPATH":
-            continue
-        cmdsize, path = lines[line_no : line_no + 2]
-        assert cmdsize.startswith("cmdsize ")
-        paths.append(RPATH_RE.match(path).groups()[0])
-        line_no += 2
-    return tuple(paths)
+    all_rpaths = _parse_otool_rpaths(otool.stdout)
+    first = list(all_rpaths.values())[0]
+    if any(first != rpaths for rpaths in all_rpaths.values()):
```

Comment:
This confused me a little - maybe a comment would help.   You are checking that the `rpaths` are all the same, for each architecture, if there are multiple architectures, and erroring out otherwise.   Maybe:

```suggestion
    # Check rpaths are the same for all architectures.
    rpaths_vals = list(all_rpaths.values())
    if (len(rpaths_vals) > 1 and 
        any(rpaths_vals[0] != rpaths for rpaths in rpaths_vals[1:])):
```

Reply:
I've refactored duplicate code into a new `_ignore_architectures` function.  It should make everything more clear.